src/hipscat/pixel_math/margin_bounding.py

```diff
@@ -0,0 +1,263 @@
+"""Utilities to build bounding boxes around healpixels that include a neighor margin."""
+
+import healpy as hp
+import numpy as np
+
+from astropy.coordinates import SkyCoord
+from regions import PixCoord, PolygonSkyRegion
+import astropy.wcs as world_coordinate_system
+import astropy.units as u
+
+from . import pixel_margins as pm
+
+
+def get_margin_scale(pixel_order, margin_threshold=0.1):
+    """Get the scale value need to expand the pixel bounding box to include the `margin_threshold`.
+
+    Args:
+        pixel_order (int): the order of the pixel to which we're calculating the margin region for.
+        margin_threshold (float): the size of the border region in degrees.
+    Returns:
+        a float representing the scale factor.
+    """
+    if margin_threshold <= 0.0:
+        raise ValueError("margin_threshold must be greater than 0.")
+
+    resolution = hp.nside2resol(2**pixel_order, arcmin=True) / 60.0
+    scale = 1 + ((2 * margin_threshold) / resolution)
+    return scale
+
+
+def get_margin_bounds_and_wcs(pixel_order, pix, scale, step=10):
+    """Get the astropy `regions.PolygonPixelRegion` and `astropy.wcs` objects
+        for a given margin bounding box scale.
+
+    Args:
+        pixel_order (int): the order of the pixel to which we're calculating the margin region for.
+        pix (int): the healpixel which we're calculating the margin region for.
+        scale (float): the scale factor to apply to the given healpixel.
+        step (int): the amount of samples of one side of the given healpixel's boundaries.
+            total samples taken = 4 * step.
+    Returns:
+        polygons (list of tuples): a list of :obj:`regions.PolygonPixelRegion`
+            and :obj:`wcs.WCS` tuples, covering the full area of the given healpixels
+            scaled up area.
+    """
+    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-statements
+    corners = hp.vec2dir(
+        hp.boundaries(2**pixel_order, pix, step=1, nest=True), lonlat=True
+    )
+
+    min_ra = corners[0][1] # western corner
+    max_ra = corners[0][3] # eastern corner
+    min_dec = corners[1][2] # southern corner
+    max_dec = corners[1][0] #  northern corner
+
+    pixel_boundaries = hp.vec2dir(
+        hp.boundaries(2**pixel_order, pix, step=step, nest=True), lonlat=True
+    )
+
+    # if the eastern corner is less than the western corner, then we've hit the
+    # ra rollover and need to normalize to 0 -> 360.
+    if max_ra < min_ra:
+        max_ra = 180. + (180. - abs(max_ra))
```

Comment:
I don't follow the logic of this transform. If you have ra's in the range [-175,-180] and [175, 180] because they go over the 180 boundary, then this would map everything to [0, 5]. So the range gets cut in half.

Shouldn't this just be `ra = ra + 360.` for anything < 0?

Reply:
changed it to be the simpler one 🤙